src/hooks/useNodeJsContributorsApi.tsx

```diff
@@ -56,108 +66,107 @@ async function getMaxContributors(): Promise<[number, number]> {
  * Trying to store cached data in localStorage in order to do less consequent requests
  */
-async function fetchRandomContributor() {
+export async function fetchRandomContributor(): Promise<Contributor> {
   let maxContributors: number | null = null;
   let fetchDate: number | null = null;
   let needToRefetch = false;
+  let contributors: Contributor[] = [];
   const ONE_MONTH_MS = 2592000000;
 
   if (window.localStorage) {
     const maxContributorsStored =
       window.localStorage.getItem('max_contributors');
     const fetchDateStored = window.localStorage.getItem('fetch_date');
 
+    contributors = JSON.parse(
+      window.localStorage.getItem('contributors') || '[]'
+    );
+
     maxContributors = maxContributorsStored
       ? parseInt(maxContributorsStored, 10)
       : null;
     fetchDate = fetchDateStored ? parseInt(fetchDateStored, 10) : null;
   }
 
-  if (fetchDate && Date.now() - fetchDate >= ONE_MONTH_MS) {
-    needToRefetch = true;
-  }
-
-  // If localStorage and data is less than 1 month old, fetch 1 time
-  try {
-    if (maxContributors && !needToRefetch) {
-      return await getContributor(
-        Math.floor(Math.random() * Math.floor(maxContributors)) + 1
-      );
+  if (fetchDate && Date.now() - fetchDate >= ONE_MONTH_MS) needToRefetch = true;
+
+  return new Promise((resolve, reject) => {
+    try {
+      if (contributors && contributors.length > 0 && !needToRefetch) {
+        const contributor = contributors.shift();
+        if (window.localStorage) {
+          localStorage.setItem('contributors', JSON.stringify(contributors));
+        }
+        resolve(contributor as Contributor);
+      }
+
+      if (maxContributors && !needToRefetch) {
+        getContributor(
+          Math.floor(Math.random() * Math.floor(maxContributors)) + 1
+        ).then(contributor => resolve(contributor));
+      }
+
+      getMaxContributors()
+        .then(([randomPage]) => getContributor(randomPage))
+        .then(contributor => {
+          if (window.localStorage) {
+            window.localStorage.setItem('fetch_date', String(Date.now()));
+          }
+          resolve(contributor);
+        });
+    } catch {
+      reject(new Error('Failed to fetch random contributor'));
     }
-
-    const [randomPage, lastPage] = await getMaxContributors();
-
-    const contributor = await getContributor(randomPage);
-
-    if (window.localStorage) {
-      window.localStorage.setItem('fetch_date', String(Date.now()));
-      window.localStorage.setItem('max_contributors', String(lastPage));
-    }
-
-    return contributor;
-  } catch {
-    return null;
-  }
+  });
 }
 
 export function useNodeJsContributorsApi(
   isVisible: boolean
-): Contributor | null {
-  const [contributor, setContributor] = useState<Contributor | null>(null);
+): Contributor | null | undefined {
+  const [contributor, setContributor] = useState<
+    Contributor | null | undefined
```

Comment:
I doubt that `undefined` is needed here 🤔 , I could see only `Contributor` and `null`
```suggestion
    Contributor | null 
```